collectors/proc.plugin/sys_class_drm.c

```diff
@@ -0,0 +1,1076 @@
+        }
+        else read_file_error(util_mem);
+
+
+        /* Do GPU and VRAM clock frequency update */
+
+        if(likely(!read_multiline_file(c->ff_clk_gpu, (char *) c->pathname_clk_gpu, &c->clk_gpu))){
+            rrddim_set_by_pointer(c->st_clk_gpu, c->rd_clk_gpu, c->clk_gpu);
+            rrdset_done(c->st_clk_gpu);
+        }
+        else read_file_error(clk_gpu);
+
+        if(likely(!read_multiline_file(c->ff_clk_mem, (char *) c->pathname_clk_mem, &c->clk_mem))){
+            rrddim_set_by_pointer(c->st_clk_mem, c->rd_clk_mem, c->clk_mem);
+            rrdset_done(c->st_clk_mem);
+        }
+        else read_file_error(clk_mem);
+
+
+        /* Do GPU memory usage update */
+
+        if(likely(!read_single_number_file(c->pathname_mem_used_vram, (unsigned long long *) &c->used_vram) && 
+                c->total_vram)){
+
+            rrddim_set_by_pointer(  c->st_mem_usage_perc_vram, 
+                                    c->rd_mem_used_perc_vram, 
+                                    c->used_vram * 10000 / c->total_vram);
+            rrdset_done(c->st_mem_usage_perc_vram);
+
+            rrddim_set_by_pointer(c->st_mem_usage_vram, c->rd_mem_used_vram, c->used_vram);
+            rrddim_set_by_pointer(c->st_mem_usage_vram, c->rd_mem_free_vram, c->total_vram - c->used_vram);
+            rrdset_done(c->st_mem_usage_vram);
+
+        }
+        else read_file_error(used_vram);
+
+        if(likely(!read_single_number_file(c->pathname_mem_used_vis_vram, (unsigned long long *) &c->used_vis_vram) &&
+                c->total_vis_vram)){
+
+            rrddim_set_by_pointer(  c->st_mem_usage_perc_vis_vram, 
+                                    c->rd_mem_used_perc_vis_vram, 
+                                    c->used_vis_vram * 10000 / c->total_vis_vram);
+            rrdset_done(c->st_mem_usage_perc_vis_vram);
+        
+            rrddim_set_by_pointer(c->st_mem_usage_vis_vram, c->rd_mem_used_vis_vram, c->used_vis_vram);
+            rrddim_set_by_pointer(c->st_mem_usage_vis_vram, c->rd_mem_free_vis_vram, c->total_vis_vram - c->used_vis_vram);
+            rrdset_done(c->st_mem_usage_vis_vram);
+            
+        }
+        else read_file_error(used_vis_vram);
+
+        if(likely(!read_single_number_file(c->pathname_mem_used_gtt, (unsigned long long *) &c->used_gtt) &&
+                c->total_gtt)){
+                    
+            rrddim_set_by_pointer(  c->st_mem_usage_perc_gtt, 
+                                    c->rd_mem_used_perc_gtt, 
+                                    c->used_gtt * 10000 / c->total_gtt);
+            rrdset_done(c->st_mem_usage_perc_gtt);
+        
+            rrddim_set_by_pointer(c->st_mem_usage_gtt, c->rd_mem_used_gtt, c->used_gtt);
+            rrddim_set_by_pointer(c->st_mem_usage_gtt, c->rd_mem_free_gtt, c->total_gtt - c->used_gtt);
+            rrdset_done(c->st_mem_usage_gtt);
+            
+        }
+        else read_file_error(used_gtt);
+
+    }
+
+    return 0;
```

Comment:
I think we should return 1 if no cards are discovered.

Reply:
Good point, I will change the code. I also need to rebase since `multi_metadata.yaml` is no longer a thing.